api-reference/docstubs/api/calendar-allowedcalendarsharingroles.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "allowedCalendarSharingRoles"
+description: ""
+author: ""
+localization_priority: Normal
+ms.prod: ""
+doc_type: apiPageType
+---
+
+# allowedCalendarSharingRoles
+
+Namespace: microsoft.graph
+
+
+
+## Permissions
+One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/concepts/permissions-reference.md).
+
+|Permission type|Permissions (from most to least privileged)|
+|:---|:---|
+|Delegated (work or school account)|**TODO: Determine scopes **|
+|Delegated (personal Microsoft account)|Not supported.|
+|Application|**TODO: Determine AppOnly scopes **|
+
+## HTTP request
+<!-- {
+  "blockType": "ignored"
+}
+-->
+``` http
+GET /me/calendar/allowedCalendarSharingRoles
+GET /users/{usersId}/calendar/allowedCalendarSharingRoles
+GET /me/calendars/{calendarId}/allowedCalendarSharingRoles
+GET /groups/{groupsId}/calendar/allowedCalendarSharingRoles
+GET /me/joinedGroups/{groupId}/calendar/allowedCalendarSharingRoles
+GET /me/messages/{messageId}/event/calendar/allowedCalendarSharingRoles
+GET /users/{usersId}/calendars/{calendarId}/allowedCalendarSharingRoles
+GET /me/calendarGroups/{calendarGroupId}/calendars/{calendarId}/allowedCalendarSharingRoles
+```
+
+## Request headers
+|Name|Description|
+|:---|:---|
+|Authorization|Bearer {token}.Required|
+
+## Request body
+In the request URL, provide the following query parameters with values.
+The following table shows the parameters that can be used with this function.
+
+|Property|Type|Description|
+|:---|:---|:---|
+|User|String||
+
+
+
+## Response
+If successful, this function returns a `200 OK` response code and a Enumeration collection in the response body.
```

Comment:
Below we seem to know the return type of collection(microsoft.graph.calendarroletype), so why couldn't we say "calenderRoleType enumeration collection" here?

Reply:
Actually we can't even reference another topic here.  So suggest:
If successful, this f unction returns a `200 OK` response code and a microsoft.graph.calendarRoleType collection, which can contain the following values: {list valid enum values}.

Reply:
Only for types and enums in a sub-namespace do we make a point to display a fully-qualified type in this kind of contexts, shall we just say:
"... and a calendarRoleType collection"?

I think to accommodate auto-generating useful (enough) docs, Dan's suggestion may be the middle way out (I'll explain below) considering the ongoing [guidance](https://msgo.azurewebsites.net/add/document/guidelines/enumerations.html), how existing enums have been documented, and what can be done with automation.

Now I'll go into ideally, according to guidance and the case of this particular function, there is one thing a current author should do manually - link calendarRoleType to its definition.

Unlike most enums, there's currently a table that defines the enum, in [calendarPermission](https://docs.microsoft.com/en-us/graph/api/resources/calendarpermission?view=graph-rest-beta) which also references the calendarRoleType enum,
Because this function has the binding parameter type as calendar (and not calendarPermissions which currently defines the enum table), the current author should move the enum table to its own enum type topic. The author should then link from both this function and calendarPermissions to that enum type topic. 
In that case, for future maintenance, we should refrain from listing the possible values in the function and calendarPermissions topics.